gtdynamics/utils/Trajectory.cpp

```diff
@@ -162,7 +161,10 @@ void Trajectory::writeToFile(const Robot &robot, const std::string &name,
   for (auto &&joint : robot.joints()) {
     jnames.push_back(joint->name());
   }
-  string jnames_str = boost::algorithm::join(jnames, ",");
+  string jnames_str = "";
+  for (size_t j = 0; j < jnames.size(); j++) {
+    jnames_str += jnames[j] + (j != jnames.size() - 1 ? "," : "");
+  }
```

Comment:
I might be being dumb but why not just construct the `jnames_str` in the first for-loop instead of constructing the `jnames` vector then doing another loop to construct the string? (and pop_back the last comma)

See also: [stackoverflow `std::copy` one-liner join](https://stackoverflow.com/a/5689042/9151520)

Reply:
Just tried to avoid any semantic changes. Someone else can update this if the efficiency is needed.

Reply:
I saw the one-liner before but it adds a delimiter at the end which needs to be removed, and you need to declare an output stringstream, so it isn't quite a one-liner.